.github/workflows/build-image-on-push.yml

```diff
@@ -0,0 +1,56 @@
+name: "Test build of image when Dockerfile is changed"
+
+on:
+  push:
+    paths:
+    - 'Dockerfile'
+    branches-ignore:
+    - main
+  pull_request:
+    paths:
+    - 'Dockerfile'
+  workflow_dispatch:
+
+jobs:
+  rebuild-container:
+    name: "Build image with cache"
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v3.0.0
+        with:
+          platforms: linux/arm64
+      - 
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.3.0
+      - 
+        name: Checkout repository
+        uses: actions/checkout@v4
+      - 
+        name: Build x86_64 image
+        id: docker_build_amd64
+        uses: docker/build-push-action@v5.3.0
+        with:
+          push: false
+          load: true
+          platforms: linux/amd64
+          tags: sentrum:amd64
+          cache-from: type=registry,ref=ghcr.io/${{ github.repository_owner }}/sentrum:latest
+      - 
+        name: Test-run x86_64 image
```

Comment:
Why only test on x86 and not on arm64?

Reply:
Sadly there aren't arm runners for Github Actions, so there is no way to test arm builds natively. The most you can do is ensure both builds complete individually, as well as test the amd64 image.

I did a lot of research on this and was disappointed there wasn't a way to test them natively that I could find!